### ISSUE TITLE
Serving combined assets is slow

### ISSUE THREAD

**Issue by aurelien-roy** [octobercms/october#5069]
- OctoberCMS Build: 465
- PHP Version: 7.3

While benchmarking my website, I noticed combined assets took some time to load. This is due to a high TTFB (>100ms), whereas "normal" (non-combined) assets has 10ms TTFB.

The reason is that the current .htaccess is configured to blacklist the `/storage/cms` folder (which contains all combined assets). Thus, every combined asset is served through October and not directly by Apache, even if the combined asset is already generated. The problem with this approach is that it requires to boot the Laravel framework (which typically takes 100ms) causing additional delays to serve these assets (as if they were directly served by Apache/Nginx).

Thus, with the exception of very slow connections, it seems that using the Asset Combiner leads to worst performance on my machine.

Hence my question: Is there an quickly achievable optimization to make all of this faster? Can the .`htaccess` be safely modified so that existing combined assets can be served directly? My guess is that would require to change the asset file naming scheme (to include timestamp infomation directly in the stored filename) but I am unsure.

Example using the same webserver:

Normal asset (TTFB 15ms)
<img width="456" alt="Capture d’écran 2020-05-10 à 16 23 12" src="https://user-images.githubusercontent.com/1875998/81501989-7a906d00-92db-11ea-936b-4e41129c2263.png">

Combined asset (TTFB 114ms, due to the framework boot time. Note that this asset has already been generated before. This is only the serving time)

<img width="435" alt="Capture d’écran 2020-05-10 à 16 32 35" src="https://user-images.githubusercontent.com/1875998/81502054-e4107b80-92db-11ea-8529-3f11058c6d28.png">

**Comment by LukeTowers** [comment]
This would somewhat tie into the approach I'm planning on taking with the resize filter: https://github.com/octobercms/october/issues/5059

**Comment by PubliAlex** [comment]
ahhh, +1 for this. Big TTFB is a problem for me since I use october CMS, with some TTFB reaching the second. I tried everything and finally opted for a static html caching (even if it bring lot of problems too).

I hope that issue is explaining why I have high TTFB without using static cache cause I always used assets combining features.

**Comment by bennothommo** [comment]
@aurelien-roy @PubliAlex not that it resolves the actual TTFB speeds of the combiner, but I've had a lot of success in just simply suffixing a `.css` or `.js` to the end of combined assets and using standard browser caching (`Cache-Control` or `Expires`).

**Comment by LukeTowers** [comment]
@bennothommo the combiner should already be setting the cache headers, I think we looked at adding the extension to the URL once before but it ended up breaking something, probably related to server configuration not passing those requests through to the combiner.

**Comment by bennothommo** [comment]
@LukeTowers interesting - I've had no problems with it myself, either during development or on production.

**Comment by LukeTowers** [comment]
Yeah, it was only a couple people that reported it as an issue, so it was more so on their side with server configuration, but it was either a common way to configure it that triggered the issue or what our docs recommended at the time so we couldn't change it.

Ideally the solution to this is to just have the combiner generate a static file just like the new resizer is going to do, but that's something to be implemented later.

**Comment by Samuell1** [comment]
@bennothommo @LukeTowers it was issue when server was thinking that file path exists on server as real file not generated
